core/src/toga/widgets/scrollcontainer.py

```diff
@@ -89,13 +89,13 @@ def content(self, widget):
         if widget:
             widget.app = self.app
             widget.window = self.window
-
             self._impl.set_content(widget._impl)
         else:
             self._impl.set_content(None)
 
         self._content = widget
-        self.refresh()
+        if widget:
+            widget.refresh()
```

Comment:
My reasoning for this change is that setting the content should not affect the size of `self` (the container), but it will affect the size of `widget` (the content).

Is this correct? And if so, is there anything which might incorrectly be depending on the previous behavior?

Reply:
Agreed that this seems correct. I can't think of any behavior that would depending on this - it's just an easy optimisation, AFAICT.

On Cocoa/iOS, calling refresh on the widget will cause `set_bounds()` to be invoked on the widget, which will trigger a refresh on the content anyway; this optimisation shortcuts the unnecessary changes to the container, and goes directly to the content. When the content layout completes (or any other content change is applied), the container scroll bounds are updated by the `on_refresh` callback on the container.

On GTK, all you're changing is which widget is marked as dirty; once the widget is marked as dirty, it will get a layout in the next render pass. Again, all this change does is optimise how much is marked as dirty.